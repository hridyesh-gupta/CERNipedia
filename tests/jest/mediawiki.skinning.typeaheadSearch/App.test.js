const VueTestUtils = require( '@vue/test-utils' );
const App = require( '../../../resources/src/mediawiki.skinning.typeaheadSearch/App.vue' );
const urlGeneratorFn = require( '../../../resources/src/mediawiki.skinning.typeaheadSearch/urlGenerator.js' );
const scriptPath = '/w/index.php';
const urlGenerator = urlGeneratorFn( scriptPath );

const defaultProps = {
	prefixClass: 'vector-',
	id: 'searchform',
	searchAccessKey: 'f',
	searchTitle: 'search',
	showThumbnail: true,
	showDescription: true,
	highlightQuery: true,
	urlGenerator,
	restClient: {
		loadMore: () => Promise.resolve(),
		fetchByTitle: () => Promise.resolve()
	},
	searchPlaceholder: 'Search MediaWiki',
	searchQuery: ''
};

const mount = ( /** @type {Object} */ customProps ) => VueTestUtils.mount( App, {
	props: Object.assign( {}, defaultProps, customProps ),
	global: {
		mocks: {
			$i18n: ( /** @type {string} */ str ) => ( {
				text: () => str
			} )
		},
		directives: {
			'i18n-html': ( el, binding ) => {
				el.innerHTML = `${ binding.arg } (${ binding.value })`;
			}
		}
	}
} );

let matchMediaResult;

describe( 'App', () => {
	beforeEach( () => {
		matchMediaResult = ( {
			matches: false
		} );
		window.matchMedia = jest.fn( () => matchMedia );
	} );
	it( 'renders a typeahead search component', () => {
		const wrapper = mount();
		expect(
			wrapper.element
		).toMatchSnapshot();
	} );
} );

describe( 'App (mobile mode)', () => {
	beforeEach( () => {
		matchMediaResult = ( {
			matches: true
		} );
		window.matchMedia = jest.fn( () => matchMediaResult );
	} );

	it( 'renders a typeahead search component in a dialog for mobile mode', () => {
		const wrapper = mount( {
			supportsMobileExperience: true,
			router: {
				addRoute: jest.fn()
			}
		} );
		expect(
			wrapper.element
		).toMatchSnapshot();

		// resize the window (but keep as desktop)
		matchMediaResult.onchange();
		expect(
			wrapper.element
		).toMatchSnapshot();
	} );
} );
