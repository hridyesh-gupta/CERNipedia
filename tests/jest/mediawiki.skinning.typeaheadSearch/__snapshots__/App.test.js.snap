// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App (mobile mode) renders a typeahead search component in a dialog for mobile mode 1`] = `<!--teleport start-->`;

exports[`App (mobile mode) renders a typeahead search component in a dialog for mobile mode 2`] = `<!--teleport start-->`;

exports[`App renders a typeahead search component 1`] = `
<div
  class="vector-typeahead-search-wrapper"
>
  
  <div
    class="cdx-typeahead-search cdx-typeahead-search--show-thumbnail vector-typeahead-search"
  >
    <form
      action=""
      class="cdx-typeahead-search__form"
      id="searchform"
    >
      <div
        class="cdx-search-input cdx-search-input--has-end-button cdx-typeahead-search__input"
      >
        <div
          class="cdx-search-input__input-wrapper"
        >
          <div
            class="cdx-text-input cdx-text-input--has-start-icon cdx-text-input--status-default cdx-search-input__text-input"
          >
            <input
              accesskey="f"
              aria-autocomplete="list"
              aria-controls="v-0"
              aria-expanded="false"
              aria-label="Search MediaWiki"
              autocomplete="off"
              class="cdx-text-input__input"
              name="search"
              placeholder="Search MediaWiki"
              role="combobox"
              search-results-label="searchresults"
              size="1"
              title="search"
              type="search"
            />
            <span
              class="cdx-icon cdx-icon--medium cdx-text-input__icon-vue cdx-text-input__start-icon"
            >
              <svg
                aria-hidden="true"
                height="20"
                viewBox="0 0 20 20"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <!--v-if-->
                <g>
                  <path
                    d="M12.2 13.6a7 7 0 111.4-1.4l5.4 5.4-1.4 1.4zM3 8a5 5 0 1010 0A5 5 0 003 8"
                  />
                </g>
              </svg>
            </span>
            <!--v-if-->
            <!--v-if-->
          </div>
          
          <div
            class="cdx-menu cdx-typeahead-search__menu"
            style="display: none;"
          >
            <ul
              aria-label="Search results"
              class="cdx-menu__listbox"
              id="v-0"
              role="listbox"
              tabindex="-1"
            >
              <!--v-if-->
              <!--v-if-->
              
              
              <!--v-if-->
            </ul>
          </div>
          
        </div>
        <button
          class="cdx-button cdx-button--action-default cdx-button--weight-normal cdx-button--size-medium cdx-button--framed cdx-search-input__end-button"
        >
          
          searchbutton
          
        </button>
      </div>
      
      <input
        name="title"
        type="hidden"
        value="Special:Search"
      />
      <input
        name="wprov"
        type="hidden"
        value="acrw1_-1"
      />
      
    </form>
  </div>
  
</div>
`;
